src/Mod/Part/App/FeatureExtrusion.cpp

```diff
@@ -337,154 +341,361 @@ App::DocumentObjectExecReturn* Extrusion::execute(void)
 
 void Extrusion::makeDraft(const ExtrusionParameters& params, const TopoDS_Shape& shape, std::list<TopoDS_Shape>& drafts)
 {
-    double distanceFwd = tan(params.taperAngleFwd) * params.lengthFwd;
-    double distanceRev = tan(params.taperAngleRev) * params.lengthRev;
+    std::vector<std::vector<TopoDS_Shape>> wiresections;
+
+    auto addWiresToWireSections =
+        [&shape](std::vector<std::vector<TopoDS_Shape>>& wiresections) -> size_t {
+        TopExp_Explorer ex;
+        size_t i = 0;
+        for (ex.Init(shape, TopAbs_WIRE); ex.More(); ex.Next(), ++i) {
+            wiresections.push_back(std::vector<TopoDS_Shape>());
+            wiresections[i].push_back(TopoDS::Wire(ex.Current()));
+        }
+        return i;
+    };
 
+    double distanceFwd = tan(params.taperAngleFwd)*params.lengthFwd;
+    double distanceRev = tan(params.taperAngleRev)*params.lengthRev;
     gp_Vec vecFwd = gp_Vec(params.dir) * params.lengthFwd;
     gp_Vec vecRev = gp_Vec(params.dir.Reversed()) * params.lengthRev;
 
     bool bFwd = fabs(params.lengthFwd) > Precision::Confusion();
     bool bRev = fabs(params.lengthRev) > Precision::Confusion();
-    bool bMid = !bFwd || !bRev || params.lengthFwd * params.lengthRev > 0.0; //include the source shape as loft section?
+    // only if there is a 2nd direction and the negated angle is equal to the first one
+    // we can omit the source shape as loft section
+    bool bMid = !bFwd || !bRev || -1.0 * params.taperAngleFwd != params.taperAngleRev;
 
-    TopoDS_Wire sourceWire;
     if (shape.IsNull())
         Standard_Failure::Raise("Not a valid shape");
-    if (shape.ShapeType() == TopAbs_WIRE) {
-        ShapeFix_Wire aFix;
-        aFix.Load(TopoDS::Wire(shape));
-        aFix.FixReorder();
-        aFix.FixConnected();
-        aFix.FixClosed();
-        sourceWire = aFix.Wire();
-    }
-    else if (shape.ShapeType() == TopAbs_FACE) {
-        TopoDS_Wire outerWire = ShapeAnalysis::OuterWire(TopoDS::Face(shape));
-        sourceWire = outerWire;
-    }
-    else if (shape.ShapeType() == TopAbs_COMPOUND) {
-        TopoDS_Iterator it(shape);
-        for (; it.More(); it.Next()) {
-            makeDraft(params, it.Value(), drafts);
+
+    // store all wires of the shape into an array
+    size_t numWires = addWiresToWireSections(wiresections);
+    if (numWires == 0)
+        Standard_Failure::Raise("Extrusion: Input must not only consist if a vertex");
+
+    // to store the sections for the loft
+    std::list<TopoDS_Wire> list_of_sections;
+    std::vector<std::vector<TopoDS_Shape>> extrusionSections;
+
+    // we need for all found wires an offset copy of them
+    // we store them in an array
+    TopoDS_Wire offsetWire;
+    for (auto& wire : wiresections)

```

Comment:
There is a warning due to unused `wire` variable.

Reply:
I need `wire ` as counter. I don't know how to silence the compiler since I don't get this warning.

Reply:
Add `(void)wire;` inside the loop.
or use `std::generate` over the resized `extrusionSections` and fill it with a lambda :
`std::vector<std::vector<TopoDS_Shape>> extrusionSections(wiresections.size());`
` std::generate(extrusionSections.begin(), extrusionSections.end(), []{return std::vector<TopoDS_Shape>();});`

Reply:
Actually, no need the loop:
`std::vector<std::vector<TopoDS_Shape>> extrusionSections(wiresections.size(), std::vector<TopoDS_Shape>());`

Reply:
Thanks! Fixed with commit 8eab4a397e